presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/UnwrapDateFunctionInPredicate.java

```diff
@@ -0,0 +1,249 @@
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.AND;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This rule rewrites predicates of the form:
+ *    date(timestamp_column) = DATE 'YYYY-MM-DD'
+ * into:
+ *    timestamp_column >= TIMESTAMP 'YYYY-MM-DD 00:00:00.000'
+ * AND timestamp_column < TIMESTAMP 'YYYY-MM-DD+1 00:00:00.000'
+ */
+public class UnwrapDateFunctionInPredicate
+        implements Rule<FilterNode>
+{
+    private static final String DATE_FUNCTION = "date";
+
+    private final FunctionAndTypeManager functionAndTypeManager;
+    private final StandardFunctionResolution functionResolution;
+
+    public UnwrapDateFunctionInPredicate(FunctionAndTypeManager functionAndTypeManager)
+    {
+        this.functionAndTypeManager = requireNonNull(functionAndTypeManager, "functionAndTypeManager is null");
+        this.functionResolution = new FunctionResolution(functionAndTypeManager.getFunctionAndTypeResolver());
+    }
+
+    @Override
+    public Pattern<FilterNode> getPattern()
+    {
+        return typeOf(FilterNode.class);
+    }
+
+    @Override
+    public Result apply(FilterNode node, Captures captures, Context context)
+    {
+        RowExpression predicate = node.getPredicate();
+        System.out.println("Original Predicate: " + predicate);
+
+        Optional<RowExpression> rewritten = rewritePredicate(predicate);
+
+        if (rewritten.isPresent()) {
+            System.out.println("Rewritten Predicate: " + rewritten.get());
+        } else {
+            System.out.println("No transformation applied.");
+        }
+
+        return rewritten.map(rowExpression -> Result.ofPlanNode(
+                new FilterNode(
+                        node.getSourceLocation(),
+                        node.getId(),
+                        node.getSource(),
+                        rowExpression))).orElseGet(Result::empty);
+    }
+
+    private Optional<RowExpression> rewritePredicate(RowExpression expression)
+    {
+        if (!(expression instanceof CallExpression)) {
+            return Optional.empty();
+        }
+
+        CallExpression call = (CallExpression) expression;
+        // Check if this is an equality comparison
+        if (!functionResolution.isEqualsFunction(call.getFunctionHandle())) {
+            return Optional.empty();
+        }
+
+        RowExpression left = call.getArguments().get(0);
+        RowExpression right = call.getArguments().get(1);
+
+        // Try both: function(column) = literal and literal = function(column)
+        Optional<RowExpression> rewritten = tryRewriteFunctionEqualsLiteral(left, right);
+        if (rewritten.isPresent()) {
+            return rewritten;
+        }
+
+        return tryRewriteFunctionEqualsLiteral(right, left);
+    }
+
+    private Optional<RowExpression> tryRewriteFunctionEqualsLiteral(RowExpression functionSide, RowExpression literalSide)
```

Comment:
We should be able to easily add some unit tests here to verify correctness of this function. Also, I think this method can be marked `static`?